src/App.jsx

```diff
@@ -14,45 +16,85 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [sortedItems, setSortedItems] = useState(goodsFromServer);
+  const [isAlphaSorted, setIsAlphaSorted] = useState(false);
+  const [isLengthSorted, setIsLengthSorted] = useState(false);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortAlpha = () => {
+    setSortedItems([...goodsFromServer].sort((a, b) => a.localeCompare(b)));
+    setIsAlphaSorted(true);
+    setIsLengthSorted(false);
+  };
+
+  const sortByLength = () => {
+    setSortedItems([...goodsFromServer].sort((a, b) => a.length - b.length));
+    setIsLengthSorted(true);
+    setIsAlphaSorted(false);
+  };
+
+  const reversed = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setSortedItems(goodsFromServer);
+    setIsLengthSorted(false);
+    setIsAlphaSorted(false);
+    setIsReversed(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info', { 'is-light': !isAlphaSorted })}
+          onClick={sortAlpha}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button', 'is-success', { 'is-light': !isLengthSorted },
+          )}
+          onClick={sortByLength}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning', { 'is-light': !isReversed })}
+          onClick={reversed}
+        >
+          Reverse
+        </button>
+
+        {(isAlphaSorted || isLengthSorted || isReversed)
+          && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={reset}
+          >
+            Reset
+          </button>
+          )
+        }
+      </div>
+
+      <ul>
+        {!isReversed && sortedItems.map(good => (
+          <li data-cy="Good" key={good}>{ good }</li>
+        ))}
+        {isReversed && sortedItems.map(good => (
+          <li data-cy="Good" key={good}>{ good }</li>
+        )).reverse()}
```

Comment:
You don't need to do it twice and also do the reverse here. Just put reverse outside in function.